google/cloud/storage/client.cc

```diff
@@ -277,43 +285,40 @@ StatusOr<ObjectMetadata> Client::UploadStreamResumable(
     auto const actual_committed_size = upload->committed_size.value_or(0);
     if (actual_committed_size != expected) {
       // Defensive programming: unless there is a bug, this should be dead code.
-      return Status(
-          StatusCode::kInternal,
-          "Mismatch in committed size expected=" + std::to_string(expected) +
-              " got=" + std::to_string(actual_committed_size) +
+      return google::cloud::internal::InternalError(
+          absl::StrCat(
+              "Mismatch in committed size expected=", expected,
+              " got=", actual_committed_size,
               ". This is a bug, please report it at "
-              "https://github.com/googleapis/google-cloud-cpp/issues/new");
+              "https://github.com/googleapis/google-cloud-cpp/issues/new"),
+          GCP_ERROR_INFO());
     }
 
     // We only update `committed_size` when uploading is successful.
     committed_size = expected;
   }
-  return Status(StatusCode::kInternal,
-                "Upload did not complete but source is exhausted");
+  return google::cloud::internal::InternalError(
+      "Upload did not complete but source is exhausted", GCP_ERROR_INFO());
 }
 
 Status Client::DownloadFileImpl(internal::ReadObjectRangeRequest const& request,
                                 std::string const& file_name) {
-  auto report_error = [&request, file_name](char const* func, char const* what,
-                                            Status const& status) {
-    std::ostringstream msg;
-    msg << func << "(" << request << ", " << file_name << "): " << what
-        << " - status.message=" << status.message();
-    return Status(status.code(), std::move(msg).str());
+  auto const* func = __func__;
+  auto msg = [&request, &file_name, func](char const* what) {
+    std::ostringstream os;
+    os << func << "(" << request << ", " << file_name << "): " << what;
+    return std::move(os).str();
   };
 
   auto stream = ReadObjectImpl(request);
-  if (stream.bad()) {
-    return report_error(__func__, "cannot open download source object",
```

Comment:
(just pointing out that) there is no longer a `msg`

Reply:
Yes, I decided to just pass the error without adding decoration to it.